overlord/snapstate/snapmgr.go

```diff
@@ -389,24 +399,24 @@ func (m *SnapManager) doPrepareSnap(t *state.Task, _ *tomb.Tomb) error {
 			panic("internal error: invalid local revision built: " + revision.String())
 		}
 		snapst.LocalRevision = revision
-		ss.Revision = revision
-		snapst.Candidate = &snap.SideInfo{Revision: ss.Revision}
+		ss.SideInfo.Revision = revision
+		snapst.Candidate = &snap.SideInfo{Revision: ss.Revision()}
```

Comment:
this looks wrong, though nothing explodes, because it's leaving out RealName, are we missing a test?

anyway after that I think it could just be:

snapst.Candidate = ss.SideInfo